test.xunit.runner.visualstudio.testadapter/RunSettingsHelperTests.cs

```diff
@@ -0,0 +1,134 @@
+
+        [Fact]
+        public void RunSettingsHelperShouldReadValuesCorrectly()
+        {
+            string settingsXml =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <DisableAppDomain>true</DisableAppDomain>
+                       <DisableParallelization>true</DisableParallelization>
+                     </RunConfiguration>
+                </RunSettings>";
+
+            RunSettingsHelper.ReadRunSettings(settingsXml);
+
+            // Correct values must be sets
+            Assert.Equal(true, RunSettingsHelper.DisableAppDomain);
+            Assert.Equal(true, RunSettingsHelper.DisableParallelization);
+
+            settingsXml =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <DisableAppDomain>false</DisableAppDomain>
+                       <DisableParallelization>true</DisableParallelization>
+                     </RunConfiguration>
+                </RunSettings>";
+
+            RunSettingsHelper.ReadRunSettings(settingsXml);
+
+            // Values must be the latest ones 
+            Assert.Equal(false, RunSettingsHelper.DisableAppDomain);
+            Assert.Equal(true, RunSettingsHelper.DisableParallelization);
+
+            settingsXml =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <DisableAppDomain>true</DisableAppDomain>
+                       <DisableParallelization>false</DisableParallelization>
+                     </RunConfiguration>
+                </RunSettings>";
+
+            RunSettingsHelper.ReadRunSettings(settingsXml);
+
+            // Values must be the latest ones 
+            Assert.Equal(true, RunSettingsHelper.DisableAppDomain);
+            Assert.Equal(false, RunSettingsHelper.DisableParallelization);
+
+
+            settingsXml =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <DisableAppDomain>false</DisableAppDomain>
+                       <DisableParallelization>false</DisableParallelization>
+                     </RunConfiguration>
+                </RunSettings>";
+
+            RunSettingsHelper.ReadRunSettings(settingsXml);
+
+            // Values must be the latest ones 
+            Assert.Equal(false, RunSettingsHelper.DisableAppDomain);
+            Assert.Equal(false, RunSettingsHelper.DisableParallelization);
+        }
+
+        [Fact]
+        public void RunSettingsHelperShouldIgnoreEvenIfAdditionalElementsExist()
+        {
+            string settingsXml =
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <TargetPlatform>x64</TargetPlatform>
```

Comment:
Consider:  For completeness, maybe add an element at the start that is unlikely to ever be part of the runsettings schema.  Example:

```<Bogus>value</Bogus>```